driver/src/contracts/stablex_contract.rs

```diff
@@ -139,40 +141,38 @@ pub trait StableXContract {
 }
 
 impl StableXContract for StableXContractImpl {
-    fn get_current_auction_index(&self) -> Result<u32> {
-        let auction_index = self.instance.get_current_batch_id().call().wait()?;
-        Ok(auction_index)
+    fn get_current_auction_index(&self) -> BoxFuture<'static, Result<u32>> {
+        let future = self.instance.get_current_batch_id().call();
```

Comment:
Is the `let future = ...` outside the `async` block because of the `'static` lifetime?

Reply:
I definitely can't put it in the async block because of compilation errors. Probably because of 'static and will go away by changing to &'a self.